### PR TITLE
Adding Audition, Essay and Speaking modules.

### PR DESCRIPTION
**Pull request by 4500zenja1**
Also adding enum LevelEnum at BaseDataModel and Permissions at Role. Right now these fixes are temporarily and could eb changed at any time.

### CODE REVIEW

File: Elevel/Domain/Models/Audition.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Elevel.Domain.Models
+{
+    public class Audition: BaseDataModel
+    {
+        public Guid AuditionId { get; set; }
+
+        public string AuditionAudio { get; set; }
+
+        public Guid QuestionID { get; set; }
```
Comment (parys) [inline Elevel/Domain/Models/Audition.cs:15]: it should be same everywhere, or ID or Id at the end.

File: Elevel/Domain/Models/Essay.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public class Essay: BaseDataModel
+    {
+        public Guid EssayId { get; set; }
+
+        public Level Level { get; set; }
+
+        public string Topic { get; set; }
+
+        public DateTime CreationDate { get; set; }
```
Comment (parys) [inline Elevel/Domain/Models/Essay.cs:17]: Please use DateTimeOffset please

File: Elevel/Domain/Models/Speaking.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public class Speaking: BaseDataModel
+    {
+        public Guid SpeakingId { get; set; }
+
+        public Level Level { get; set; }
+
+        public string Topic { get; set; }
+
+        public DateTime CreationDate { get; set; }
```
Comment (parys) [inline Elevel/Domain/Models/Speaking.cs:17]: Please use DateTimeOffset please

**Comment by 4500zenja1** [comment]
A new push request without (maybe?) problems concerning the same issue was created, this push request is closed now.

**Pull request closed without merging**